src/components/tab-side-menu/tab-side-menu.tsx

```diff
@@ -60,27 +63,24 @@ const topHeaderStyle = cva("flex w-full items-center justify-between", {
 export default function TabSideMenu() {
   const [isTabMinimized, setIsTabMinimized] = useState(false);
   const [isAdding, setIsAdding] = useState(false);
-  const [goals, setGoals] = useState<TabItem[]>([]);
   const [newGoal, setNewGoal] = useState("");
+  const { mutate } = useCreateGoal();
+  const { data } = useSuspenseInfiniteQuery(getInfiniteGoalsOptions({}));
```

Comment:
size 옵션을 지정해줘야 할 것 같습니다. 지금은 목표가 많아져도 기본 사이즈가 3개라 초기 로딩 시 3개의 목표만 보이는 문제가 있습니다. 디바이스의 높이에 따라 size를 정할 수 있으면 좋을 것 같은데, 일단은 최소 20개는 넘는 사이즈로 지정해 두는 게 어떨까요?

```suggestion
  const { data } = useSuspenseInfiniteQuery(getInfiniteGoalsOptions({ size: 20 }));
```

Reply:
그리고 무한스크롤 기능을 위해 InfiniteQuery를 쓰신 것 같은데, 현재 다음 페이지를 가져오는 로직이 누락되어 있습니다. 목표 리스트의 끝까지 스크롤이 되었을 때 다음 페이지를 가져오도록 수정 부탁드립니다!